src/lib/mina_base/test/zkapp_command/verifiable.ml

```diff
@@ -0,0 +1,140 @@
+open Core_kernel
+open Signature_lib
+
+let call_forest_gen = Zkapp_command_test.Call_forest.gen
+
+open Mina_base
+open Zkapp_command
+open Zkapp_command.Verifiable
+
+let check_verifiable_property () =
+  let processed_account_ids = ref Account_id.Map.empty in
+  let find_vk_create vk_hash account_id =
+    let () =
+      processed_account_ids :=
+        Account_id.Map.set !processed_account_ids ~key:account_id ~data:vk_hash
+    in
+    Ok { With_hash.data = Side_loaded_verification_key.dummy; hash = vk_hash }
+  in
+  let find_vk_check vk_hash account_id =
+    match Account_id.Map.find !processed_account_ids account_id with
+    | Some vk_hash' ->
+        if Zkapp_basic.F.equal vk_hash' vk_hash then
+          Ok
+            { With_hash.data = Side_loaded_verification_key.dummy
+            ; hash = vk_hash
+            }
+        else Error (Error.of_string "Verification key hash mismatch")
+    | None ->
+        Error (Error.of_string "Verification key not found")
+  in
+  let gen_authorization_kind account_id authorization =
+    let open Quickcheck.Generator.Let_syntax in
+    let vk_hash = Zkapp_basic.F.random () in
+    match authorization with
+    | Control.None_given ->
+        Quickcheck.Generator.return Account_update.Authorization_kind.None_given
+    | Proof p ->
+        Quickcheck.Generator.return
+          (Account_update.Authorization_kind.Proof vk_hash)
+    | Signature _ ->
+        Quickcheck.Generator.return Account_update.Authorization_kind.Signature
+  in
+  let gen_account_upd =
+    let open Quickcheck.Generator.Let_syntax in
+    let%bind body = Account_update.Body.gen
```

Comment:
Looks like these will never generate vk updates. You have to pass `zkapp_account:true` 
https://github.com/MinaProtocol/mina/blob/d97cdc46600e3db5086ccbd9c7a89013fcc3bc95/src/lib/mina_base/account_update.ml#L704
Also, they are always generated with the same vk hash. A dummy vk generator would be useful for this test